flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/casting/CastRulesTest.java

```diff
@@ -236,6 +471,35 @@
                                 RawValueData.fromObject(
                                         LocalDateTime.parse("2020-11-11T18:08:01.123")),
                                 StringData.fromString("2020-11-11T18:08:01.123")),
+                CastTestSpecBuilder.testCastTo(DECIMAL(5, 3))
+                        .fromCase(
+                                DECIMAL(4, 3),
+                                DecimalData.fromBigDecimal(new BigDecimal("9.87"), 4, 3),
+                                DecimalData.fromBigDecimal(new BigDecimal("9.870"), 5, 3))
+                        .fromCase(
+                                TINYINT(),
+                                (byte) -1,
+                                DecimalData.fromBigDecimal(new BigDecimal("-1.000"), 5, 3))
```

Comment:
introduce a help method for the very repetitive `DecimalData.fromBigDecimal(new BigDecimal("-1.000"), 5, 3)`?

Reply:
This number is used only once in our tests :) Perhaps you're referring to another number?

Reply:
What you could have done is simply `decimal("-1.000")` or `decimal("-1.000", 5, 3)` and abstract away the verbose `DecimalData.fromBigDecimal(new BigDecimal("-1.000"), 5, 3)` for readbility.